src/app/features/notification/notification.component.ts

```diff
@@ -35,80 +37,157 @@ import { closeOnlySubmenu } from "./close-only-submenu";
+   */
+  private filterUserNotifications(notifications: NotificationEvent[]) {
     this.allNotifications = notifications.filter(
-      (notification) => notification.sentBy === user,
+      (notification) => notification.notificationFor === this.userId,
     );
     this.unreadNotifications = notifications.filter(
       (notification) =>
-        notification.sentBy === user && !notification.readStatus,
+        notification.notificationFor === this.userId &&
+        !notification.readStatus,
     );
   }
 
-  markAllRead() {
-    // TODO: Implement the logic to mark all notifications as read.
-    Logging.log("All notifications marked as read");
+  /**
+   * Marks all notifications as read.
+   */
+  async markAllRead(): Promise<void> {
+    const unreadNotifications = this.allNotifications.filter(
+      (notification) => !notification.readStatus,
+    );
+    await this.updateReadStatusForNotifications(unreadNotifications, true);
   }
 
-  enableNotificationForUser() {
-    // TODO: Need to implement the logic so that the notification is enabled corresponding to the user.
-    Logging.log("Notification enabled");
+  /**
+   * Toggles the notification setting for the user.
+   */
+  async enableNotificationForUser(): Promise<void> {
+    // TODO: Implement the logic to called the getToken function from the NotificationService file, Once the PR #2692 merged.
+    this.hasNotificationEnabled = !this.hasNotificationEnabled;
+
+    const notificationConfig = await this.loadNotificationConfig(this.userId);
+
+    // TODO: Currently, email notification are disabled. Update this logic once the email notification feature is implemented.
+    const notificationChannels = {
+      push: this.hasNotificationEnabled,
+      email: false,
+    };
+
+    if (notificationConfig) {
+      notificationConfig.channels = notificationChannels;
+      await this.entityMapper.save(notificationConfig);
+    } else {
+      const newNotificationConfig = new NotificationConfig(this.userId);
+      newNotificationConfig.channels = notificationChannels;
+      await this.entityMapper.save(newNotificationConfig);
+    }
+
+    this.alertService.addInfo(
+      `Notification ${this.hasNotificationEnabled ? "enabled" : "disabled"}.`,
+    );
+  }
+
+  /**
+   * Updates the read status for multiple notifications.
+   */
+  private async updateReadStatusForNotifications(
+    notifications: NotificationEvent[],
+    newStatus: boolean,
+  ) {
+    for (const notification of notifications) {
+      notification.readStatus = newStatus;
+      await this.entityMapper.save(notification);
+    }
+    this.filterUserNotifications(this.allNotifications);
   }
 
+  /**
+   * Updates the read status of a single notification.
+   */
   async updateReadStatus(notification: NotificationEvent, newStatus: boolean) {
     notification.readStatus = newStatus;
     await this.entityMapper.save(notification);
     this.filterUserNotifications(this.allNotifications);
   }
```

Comment:
instead of another method, we can just call `updateReadStatusForNotifications([notification])` (wrapped in array instead of passing single item as parameter) and avoid the code duplication.